examples/libadwaita/toast.rs

```diff
@@ -0,0 +1,117 @@
+};
+use relm4::prelude::*;
+
+struct App {
+    activated: &'static str,
+    toast_overlay: adw::ToastOverlay,
+}
+
+#[derive(Debug)]
+enum Msg {
+    Activate,
+    Cancel,
+}
+
+#[relm4::component]
+impl SimpleComponent for App {
+    type Init = ();
+    type Input = Msg;
+    type Output = ();
+
+    view! {
+        adw::Window {
+            set_title: Some("Simple app"),
+            set_default_size: (300, 100),
+
+            #[name = "toast_overlay"]
+            adw::ToastOverlay {
+                gtk::Box {
+                    set_orientation: gtk::Orientation::Vertical,
+                    set_spacing: 5,
+                    set_margin_all: 5,
+                    set_valign: gtk::Align::Center,
+
+                    gtk::Button {
+                        set_hexpand: false,
+                        set_vexpand: false,
+                        set_label: "Activate",
+                        connect_clicked => Msg::Activate,
+                    },
+                    gtk::Button {
+                        set_hexpand: false,
+                        set_vexpand: false,
+                        set_label: "Cancel",
+                        connect_clicked => Msg::Cancel,
+                    },
+                    gtk::Text {
+                        #[watch]
+                        set_text: model.activated,
+                    }
+                }
+
+            }
+        }
+    }
+
+    // Initialize the component.
+    fn init(
+        _: Self::Init,
+        root: &Self::Root,
+        sender: ComponentSender<Self>,
+    ) -> ComponentParts<Self> {
+        let mut model = App {
+            activated: "Idle",
+            toast_overlay: adw::ToastOverlay::new(),
+        };
+
+        struct GroupName {}
+        impl ActionGroupName for GroupName {
+            const NAME: &'static str = "toast";
+        }
+        struct Name;
+        impl ActionName for Name {
+            type Group = GroupName;
+            type Target = ();
+            type State = ();
+            const NAME: &'static str = "cancel";
+        }
+
+        // Insert the code generation of the view! macro here
+        let widgets = view_output!();
+        let mut actions = RelmActionGroup::<GroupName>::new();
+        let action = RelmAction::<Name>::new_stateless(|_| {});
+        actions.add_action(action);
+        actions.register_for_widget(&widgets.toast_overlay);
+
+        model.toast_overlay = widgets.toast_overlay.clone();
+
+        ComponentParts { model, widgets }
+    }
+
+    fn update(&mut self, msg: Self::Input, sender: ComponentSender<Self>) {
+        match msg {
+            Msg::Activate => {
+                self.activated = "Active";
+                let toast = adw::Toast::new("Activated");
+                toast.set_action_name(Some("toast.cancel"));
+                toast.set_button_label(Some("Cancel"));
+                toast.set_timeout(0);
+                toast.connect_button_clicked(move |this| {
```

Comment:
Here I do already use the `connect_button_clicked`, but unless I do all the activation stuff for the action the button show up grey-ish like if it's deactivated and can't be clicked.

Reply:
Weird. I'll look into this myself. I don't think that's supposed to happen because otherwise the method would be kind of pointless.

Reply:
I don't see a difference, to be honest. It looks correct for me regardless of whether actions where used or not.

Without actions:

![Screenshot from 2023-10-26 23-15-11](https://github.com/Relm4/Relm4/assets/22240637/81fee202-dc31-4ddc-83ee-944df1402d06)

With actions:

![Screenshot from 2023-10-26 23-15-46](https://github.com/Relm4/Relm4/assets/22240637/91792e51-aa19-4d8d-91f9-a1626c07ede3)

I think it's just the theme that makes it look disabled because of the low contrast. The button in the Adwaita Demo app also looks like this:

![image](https://github.com/Relm4/Relm4/assets/22240637/718885b4-1aac-44de-969e-3512dcafbf70)

Reply:
Light theme with action stuff:

![image](https://github.com/Relm4/Relm4/assets/51865119/09e667c2-bab4-4b18-bcb0-df72fbbfe5b3)

Light theme without action stuff, **the button can't be clicked**:

![image](https://github.com/Relm4/Relm4/assets/51865119/411500b1-984f-4811-a729-980a2719efeb)


Dark theme with action stuff:

![image](https://github.com/Relm4/Relm4/assets/51865119/ebc5fc3d-6451-4d9f-8877-c5b1986ad663)

Dark theme without action stuff:

![image](https://github.com/Relm4/Relm4/assets/51865119/8ec0650a-c5ea-43ff-9f65-486df18ebe11)


I'm on Arch with  libadwaita 1:1.4.0-1, gtk4 1:4.12.3-2

Reply:
Weird, I also have libadwaita-1.4.0-1 (but gtk4 4.12.3-**1**) on Fedora 39 and even if I remove the entire action stuff and the signal handler (leaving just the label and the timeout) the button is sensitive and I can click it. I don't think you're doing something wrong, this seems to be an libadwaita issue...

Reply:
I will report it upstream then.

Reply:
Ok, I actually got it wrong. I forgot to remove the `set_action_name`. Now it works properly.